client/src/app/services/video.service.ts

```diff
@@ -0,0 +1,85 @@
+import { ElementRef, Injectable } from '@angular/core';
+import Hls from 'hls.js';
+import { errorLinkHandler } from '../shared/error-link.handler';
+
+const DUMMY_LINKS = [
+  'https://wisey.app/videos/think-creatively-solve-problems-easily/lesson-5/AppleHLS1/lesson-5.m3u8',
+  'https://wisey.app/videos/think-creatively-solve-problems-easily/lesson-3/AppleHLS1/lesson-3.m3u8',
+  'https://wisey.app/videos/think-creatively-solve-problems-easily/lesson-4/AppleHLS1/lesson-4.m3u8',
+];
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class VideoService {
+  public videoRef?: ElementRef<HTMLVideoElement>;
+  public hlsElement?: Hls;
+  public videoStreamError?: any
+  private videoId!: string;
+  private videoLink?: string;
+
+  constructor() { }
+
+  public setVideoId(id: string): void {
+    this.videoId = id;
+  }
+
+  public getVideoId(): string {
+    return this.videoId;
+  }
+
+  public setVideoLink(link: string): void {
+    this.videoLink = link;
+  }
+
+  public getVideoLink(): string | undefined {
+    return this.videoLink;
+  }
+
+  public async runVideoStream(link: string, ref: ElementRef<HTMLVideoElement>, currentTime: number): Promise<void> {
+    this.videoStreamError = false;
+    const response = await errorLinkHandler(link);
+    if (response) {
+      this.playVideo(link, ref, currentTime);
+    } else {
+      // Change link (small hack) to play video.
+      const alternativeLink = this.simulateDifferentLinks();
+      this.playVideo(alternativeLink, ref, currentTime);
+      // To handle video stream errors, delete 2 lines above and uncomment 2 lines below:
+      // this.videoRef?.nativeElement.remove();
+      // this.videoStreamError = true;
```

Comment:
I think it's better to handle errors instead of showing dummy data that can be related to another course

Reply:
I've done it on purpose, because almost all videos crashed (API link errors). If the link is good, this function handle errors. You are right, this could be rewrite in better way, if fix API.